mathics/core/formatter.py

```diff
@@ -509,7 +510,9 @@ def do_format_complex(element, evaluation, form) -> "Expression":
     return do_format_expression(Expression(SymbolHoldForm, result), evaluation, form)
 
 
-def do_format_expression(element, evaluation, form):
+def do_format_expression(
+    element: BaseElement, evaluation: Evaluation, form: Symbol
+) -> BaseElement:
     # not sure how much useful is this format_cache
```

Comment:
Let's simply for now then and if we find it needed, then add caching back in via one of the more conventional Python methods like 2526d0a29c7e28ccae99c4cb2ae0bedef22c1a6f

Reply:
OK, that could work. The question is if we gain something by caching this kind of functions...

Reply:
With respect to this question, what I noticed on the Expression cache where specific examples of things were given is that by reducing unnecessary conversions and evaluations and adding other specific targeted and simply done caching such as we do for sympy or mpmath functions, reduces the need and benefit of this homegrown, hard-to-understand and potentially buggy (in that it may be memory leak forming) caching mechanism. 

There is an should be a lesson here: if you don't describe and document what you are doing, someone may come along in the future and remove it or replace it with something simpler. So when we add caches and stuff like this, we also need to describe _why_ with specific examples and tests. 

Currently we are or should be benchmarking performance as we go a long. When there is a regression, we should undertake to fix that.

Reply:
The question is: how often do we call more than once a formatting function over the same expression? My guess is that almost never. This is why removing this cache didn´t have any impact on performance.